common/sgx/verifier.c

```diff
@@ -800,13 +825,30 @@ oe_result_t oe_sgx_verify_evidence(
             &sgx_endorsements));
 
         // Verify the quote now.
-        OE_CHECK(oe_verify_quote_with_sgx_endorsements(
-            report_body,
-            report_body_size,
-            &sgx_endorsements,
-            time,
-            &valid_from,
-            &valid_until));
+        OE_CHECK_NO_TCB_LEVEL(
+            result_verify_quote,
+            oe_verify_quote_with_sgx_endorsements(
+                report_body,
+                report_body_size,
+                &sgx_endorsements,
+                time,
+                &platform_tcb_level,
+                &valid_from,
+                &valid_until));
+
+        // A successful oe_verify_quote_with_sgx_endorsements() returns either
+        // OE_OK, or OE_TCB_LEVEL_INVALID as OE does not terminate verification
+        // due to TCB status not being up to date. So, the verification result
+        // should be consistent with the retrieved TCB status.
+        if ((result_verify_quote == OE_OK) !=
+            (platform_tcb_level.status.fields.up_to_date == 1))
+        {
+            OE_RAISE_MSG(
+                OE_CONSTRAINT_FAILED,
+                "Inconsistent TCB status: verify quote(%s), tcb status(%u)",
+                oe_result_str(result_verify_quote),
+                oe_tcb_level_status_to_type(&platform_tcb_level.status));
```

Comment:
Can this TCB level be logged as a string?

Reply:
Sure. Added a helper function `oe_sgx_tcb_status_str()`